src/test/components/MenuButtons.test.js

```diff
@@ -30,59 +73,124 @@ describe('app/MenuButtons', function() {
+    it('the MenuButtons buttons with profile.meta.networkURLsRemoved set to undefined', () => {
+      const profile = processProfile(createGeckoProfileWithJsTimings());
+      profile.meta.networkURLsRemoved = undefined;
+      const { store, view } = setup(profile);
+      store.dispatch(startSymbolicating());
 
-    store.dispatch(doneSymbolicating());
-    expect(profileSharing).toMatchSnapshot();
+      view.update();
+      expect(view).toMatchSnapshot();
+
+      store.dispatch(doneSymbolicating());
+      view.update();
+      expect(view).toMatchSnapshot();
+    });
+
+    it('the Privacy Notice panel when Share button is clicked', () => {
+      const { store, view, getShareButtonInput } = setup();
+      store.dispatch(doneSymbolicating());
+      getShareButtonInput().simulate('click');
+
+      view.update();
+      expect(view).toMatchSnapshot();
+    });
   });
 
-  it('renders the MenuButtons buttons with profile.meta.networkURLsRemoved set to undefined', () => {
-    const profile = processProfile(createGeckoProfileWithJsTimings());
-    profile.meta.networkURLsRemoved = undefined;
-    const store = storeWithProfile(profile);
-    store.dispatch(startSymbolicating());
+  describe('should share the profile', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
 
-    const profileSharing = renderer.create(
-      <Provider store={store}>
-        <MenuButtons />
-      </Provider>,
-      { createNodeMock }
-    );
+    beforeAll(function() {
+      (window: any).TextEncoder = TextEncoder;
+    });
+
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
+
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
+
+    it('on a button click', async () => {
+      const {
+        store,
+        view,
+        getShareButton,
+        getSecondaryShareButtonInput,
+      } = setup();
+      // Sharing without URLs
+      await getShareButton()
+        .instance()
+        ._attemptToShare();
+
+      view.update();
+      expect(view).toMatchSnapshot();
+
+      const profileSharingStatus = ProfileViewSelectors.getProfileSharingStatus(
+        store.getState()
+      );
+      expect(profileSharingStatus).toEqual({
+        sharedWithUrls: false,
+        sharedWithoutUrls: true,
+      });
+
+      // Sharing with URLs this time
+      getSecondaryShareButtonInput().simulate('click');
+      await getShareButton()
+        .instance()
+        ._attemptToSecondaryShare();
```

Comment:
I had to do it that way because `_attemptToShare` method is async. That's why we have to wait until it finishes and I couldn't find a way to determine when an async click function finishes in enzyme. So I changed `_attemptToShare` to return a Promise and used `await` it here.